### PR TITLE
transfer workload include burns

### PR DESCRIPTION
**Pull request by ptrus**
- adds staking.Burn transactions to the transfer workload
- extracts repeated "estimateGas, fund caller, sign tx, submit tx" action into a helper function

### CODE REVIEW

File: go/oasis-node/cmd/debug/txsource/workload/workload.go [hunk 1]
```diff
@@ -33,6 +33,60 @@ func FundAccountFromTestEntity(ctx context.Context, logger *logging.Logger, cnsc
...
+	// Fund caller to cover transaction fees.
+	feeAmount := int64(tx.Fee.Gas) * gasPrice
+	if err := tx.Fee.Amount.FromInt64(feeAmount); err != nil {
+		return fmt.Errorf("fee amount from int64: %w", err)
+	}
+	if err := transferFunds(ctx, logger, cnsc, fundingAccount, caller.Public(), feeAmount); err != nil {
+		return fmt.Errorf("account funding failure: %w", err)
+	}
+
+	// Submit tx.
```
Comment (pro-wh) [inline go/oasis-node/cmd/debug/txsource/workload/workload.go]:
```suggestion
	// Sign tx.
```

File: go/oasis-node/cmd/debug/txsource/workload/transfer.go [hunk 2]
```diff
@@ -21,10 +21,9 @@ const (
 	// NameTransfer is the name of the transfer workload.
 	NameTransfer = "transfer"
 
-	transferNumAccounts  = 10
-	transferAmount       = 1
-	transferFundInterval = 10
```
Comment (pro-wh) [inline go/oasis-node/cmd/debug/txsource/workload/transfer.go]: :information_source: the fate of this is that we now fund for every transaction, via a new helper

File: go/oasis-node/cmd/debug/txsource/workload/transfer.go [hunk 3]
```diff
@@ -21,10 +21,9 @@ const (
...
-	transferGasCost      = 10
```
Comment (pro-wh) [inline go/oasis-node/cmd/debug/txsource/workload/transfer.go]: :information_source: the fate of this is that the new helper does a proper gas estimation for each transaction

**Comment by ptrus** [comment]
> hm, is there some other kind of description somewhere we can use better to indicate that the "transfer" workload is 50-50 transfers and burns?

added a comment to the name constant since i couldn't come up with a workload name better indicating it

**Pull request merged**